src/Accelerators/NNPA/Conversion/ONNXToZHigh/RewriteONNXForZHigh.cpp

```diff
@@ -224,9 +224,13 @@ bool canInferencePadsForNNPAConv(ONNXConvOp op) {
   ONNXConvOpAdaptor operandAdaptor = ONNXConvOpAdaptor(op);
   ONNXConvOpShapeHelper shapeHelper(&op);
   assert(succeeded(shapeHelper.computeShape(operandAdaptor)));
+  RankedTensorType inputType = op.X().getType().cast<RankedTensorType>();
+  ArrayRef<int64_t> inputShape = inputType.getShape();
   return (shapeHelper.pads.size() == 4) &&
-         (llvm::all_of(
-             shapeHelper.pads, [](IndexExpr val) { return val.isLiteral(); }));
+         (llvm::all_of(shapeHelper.pads,
+             [](IndexExpr val) { return val.isLiteral(); })) &&
+         (op.auto_pad() != "VALID") && (inputShape.size() == 4) &&
```

Comment:
Should it be `VALID_PADDING`? Better to use `op.auto_pad(). equals_insensitive("VALID_PADDING")` instead of `!=`

Reply:
I updated the code to use equals_insensitive instead of !=. Thank you for confirmation.

In the onnx dialect, auto_pad must be either NOTSET, SAME_UPPER, SAME_LOWER or VALID (not VALID_PADDING). (c.f. https://github.com/onnx/onnx/blob/main/docs/Operators.md#Conv)